### ISSUE TITLE
undo not undoing past last save since VSCode 1.73

### ISSUE THREAD

**Issue by slnc** [VSCodeVim/Vim#8188]
**Describe the bug**
The undo command 'u' doesn't undo past the last file save anymore starting from VSCode 1.73

**To Reproduce**

1.  Modify file
2.  Save changes
3.  Press 'u' to undo changes
4.  Nothing happens (undo doesn't undo)

**Expected behavior**
Undo should undo past the last file save.

**Environment (please complete the following information):**

- Extension (VsCodeVim) version: v1.24.3
- VSCode version: v1.73.1, v.1.74.2
- OS: MacOS Monterey 12.6.2

**Additional context**
If I type cmd+z, changes do get undone past the last file save.

**Comment by rivaldi8** [comment]
Looks like a duplicate of #8157.

**Comment by slnc** [comment]
> Looks like a duplicate of #8157.

I wasn't sure because there is no mention of "Save", but if it's better that I close this one, please let me know.

I don't know if it's related, but I've also noticed a few times already some strange behavior with the VSCode + vim editor which appears correlated with this issue (and #8157). It's as if the editor contents get "corrupted": large blocks of text dissappear or get moved around and no amount of undo or redo fixes it. I have always used "u" to undo in those cases, I will try cmd+z next time. I'm sorry but I can't give a better description of steps to reproduce it.

**Comment by hitzhangjie** [comment]
The newest version still could reproduce this problem:

```
Version: 1.75.1
Commit: 441438abd1ac652551dbe4d408dfcec8a499b8bf
Date: 2023-02-08T21:34:01.965Z (1 day ago)
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Darwin arm64 22.3.0
Sandboxed: No
```

**Comment by Yves-air** [comment]
Same here. I haven't had time to track all conditions where `u` and CTRL-R does not work properly, but defintely a save stops it.
I have to revert to CTRL-Z in those cases, but CTRL-Y doesn't work as a redo, it works the proper way of vim, so I have to use the menu to redo!

**Comment by ddnomad** [comment]
@Yves-air Fix from https://github.com/VSCodeVim/Vim/issues/8157#issuecomment-1400217493 seems to work for some people, including myself. Not ideal but better than nothing.

**Comment by ddnomad** [comment]
> That's a workaround, not a fix ;)

@Yves-air Fair enough, I concur :)

**Comment by ianchanning** [comment]
> Same here. I haven't had time to track all conditions where `u` and CTRL-R does not work properly, but defintely a save stops it. I have to revert to CTRL-Z in those cases, but CTRL-Y doesn't work as a redo, it works the proper way of vim, so I have to use the menu to redo!

@Yves-air You can delete the `ctrl+y` mapping if you don't use it - then it will work again. That was the best solution I've had so far.

**Comment by ianchanning** [comment]
@Yves-air <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Z</kbd> does redo too

![Screenshot from 2023-02-10 18-10-35](https://user-images.githubusercontent.com/233318/218153884-7f12ad32-b0ff-46bd-9fb6-02bb142b5404.png)

**Comment by Yves-air** [comment]
@ianchanning 
CTRL-SHIFT-Z doesn't work for me.

In any case, I have my workaround, I just wanted to chime in here so that eventually the undo gets fixed.

**Comment by yamakoud** [comment]
> Same here. I haven't had time to track all conditions where `u` and CTRL-R does not work properly, but defintely a save stops it. I have to revert to CTRL-Z in those cases, but CTRL-Y doesn't work as a redo, it works the proper way of vim, so I have to use the menu to redo!
> 
> @Yves-air You can delete the Ctrl+Y mapping if you don't use it - then it will work again. That was the best solution I've had so far.

After I removed Ctrl+Y mapping and reload window, vim undo start work well.
Thanks!

**Comment by J-Fields** [comment]
Duplicate of https://github.com/VSCodeVim/Vim/issues/8157, which should be fixed in `v1.25.1`
